### PR TITLE
feat(config): add presets for the node config

### PR DESCRIPTION
**Pull request by yoavGrs**
## Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

-

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

## Other information

- - -
This change is [Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/1170)

### CODE REVIEW

**Review by OmriEshhar1 — CHANGES_REQUESTED** [review]
> Reviewed 6 of 6 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/1170)* status: all files reviewed, 4 unresolved discussions (waiting on @yoavGrs)

*[a discussion](https://reviewable.io/reviews/starkware-libs/papyrus/1170#-NeIckn2BYcLi2duEfrj:-NeIckn2BYcLi2duEfrk:by498tb) (no related file):*
Is there a reason why we don't use the default ingress and pvc parameters in the default values.yaml file?

___
___
*[`deployments/helm/values.yaml` line 14 at r2](https://reviewable.io/reviews/starkware-libs/papyrus/1170#-NeIbUrL849Ssd3VUb2E:-NeIbUrL849Ssd3VUb2F:b-lqzp0h) ([raw file](https://github.com/starkware-libs/papyrus/blob/490b580430b3088ec73270c76628e2be63192b06/deployments/helm/values.yaml#L14)):*
> ```YAML
> 
> starknet:
>   # possible values: "...miannet.json", "...testnet1.json", "...testnet2.json".
> ```

mainnet



_Code quote:_
```YAML
miannet
```
___
*[`deployments/helm/values.yaml` line 15 at r2](https://reviewable.io/reviews/starkware-libs/papyrus/1170#-NeIbY_h398kueKBbPNR:-NeIbY_h398kueKBbPNS:b-lqzp0h) ([raw file](https://github.com/starkware-libs/papyrus/blob/490b580430b3088ec73270c76628e2be63192b06/deployments/helm/values.yaml#L15)):*
> ```YAML
> starknet:
>   # possible values: "...miannet.json", "...testnet1.json", "...testnet2.json".
>   preset: app/config/presets/miannet.json
> ```

mainnet



_Code quote:_
```YAML
miannet
```
___
*[`deployments/helm/values.yaml` line 15 at r2](https://reviewable.io/reviews/starkware-libs/papyrus/1170#-NeIcKkr4ha18wgY1eiz:-NeIcKkr4ha18wgY1ej-:bbas39p) ([raw file](https://github.com/starkware-libs/papyrus/blob/490b580430b3088ec73270c76628e2be63192b06/deployments/helm/values.yaml#L15)):*
> ```YAML
> starknet:
>   # possible values: "...miannet.json", "...testnet1.json", "...testnet2.json".
>   preset: app/config/presets/miannet.json
> ```

if the file location is always going to be under app/config/presets, maybe it's better to keep it hard-coded in the code and have only the file name (without the full path) in the configuration.

**Review by yoavGrs — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/1170)* status: 4 of 6 files reviewed, 4 unresolved discussions (waiting on @eranreshef-starkware and @OmriEshhar1)

*[a discussion](https://reviewable.io/reviews/starkware-libs/papyrus/1170#-NeIckn2BYcLi2duEfrj:-NecdIMD-Sjr7L71E_OJ:b-rdlj3b) (no related file):*
<details><summary><i>Previously, OmriEshhar1 wrote…</i></summary><blockquote>

Is there a reason why we don't use the default ingress and pvc parameters in the default values.yaml file?
</blockquote></details>

@eranreshef-starkware 

___
___
*[`deployments/helm/values.yaml` line 14 at r2](https://reviewable.io/reviews/starkware-libs/papyrus/1170#-NeIbUrL849Ssd3VUb2E:-NecYs638yls_9r3l8DE:b-896fix) ([raw file](https://github.com/starkware-libs/papyrus/blob/490b580430b3088ec73270c76628e2be63192b06/deployments/helm/values.yaml#L14)):*
<details><summary><i>Previously, OmriEshhar1 wrote…</i></summary><blockquote>

mainnet
</blockquote></details>

Done.
___
*[`deployments/helm/values.yaml` line 15 at r2](https://reviewable.io/reviews/starkware-libs/papyrus/1170#-NeIbY_h398kueKBbPNR:-NecYsU54C9_ApMVlsRm:b-896fix) ([raw file](https://github.com/starkware-libs/papyrus/blob/490b580430b3088ec73270c76628e2be63192b06/deployments/helm/values.yaml#L15)):*
<details><summary><i>Previously, OmriEshhar1 wrote…</i></summary><blockquote>

mainnet
</blockquote></details>

Done.
___
*[`deployments/helm/values.yaml` line 15 at r2](https://reviewable.io/reviews/starkware-libs/papyrus/1170#-NeIcKkr4ha18wgY1eiz:-NecZYGx8pMjaAR9xW7f:b-896fix) ([raw file](https://github.com/starkware-libs/papyrus/blob/490b580430b3088ec73270c76628e2be63192b06/deployments/helm/values.yaml#L15)):*
<details><summary><i>Previously, OmriEshhar1 wrote…</i></summary><blockquote>

if the file location is always going to be under app/config/presets, maybe it's better to keep it hard-coded in the code and have only the file name (without the full path) in the configuration.
</blockquote></details>

Done.

**Review by OmriEshhar1 — COMMENTED** [review]
> Reviewed 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/1170)* status: all files reviewed, 1 unresolved discussion (waiting on @eranreshef-starkware)

**Comment by eranreshef-starkware** [comment]
<details><summary><i>Previously, yoavGrs wrote…</i></summary><blockquote>

@eranreshef-starkware
</blockquote></details>

ingress and storage are "deployment-specific" parameters. since papyrus is a public repo, there is no point in setting as defaults the specific values we use in starkware.

**Review by OmriEshhar1 — DISMISSED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/1170#-:-NehXLI64oxt-BRXmPBz:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @yoavGrs)

**Review by OmriEshhar1 — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/1170)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @yoavGrs)

File: deployments/helm/templates/deployment.yaml [hunk 1]
```diff
@@ -58,10 +58,8 @@ spec:
             memory: {{ .Values.deployment.resources.requests.memory }}
         {{- if not .Values.backup.enabled }}
         args:
-        - --chain_id
-        - {{ .Values.starknet.chainId }}
-        - --central.url
-        - {{ .Values.starknet.centralUrl }}
+        - --config_file
+        - /app/config/presets/{{ .Values.starknet.preset }}
```
Comment (eranreshef-starkware) [inline deployments/helm/templates/deployment.yaml:62]: [backup.sh](https://github.com/starkware-libs/papyrus/blob/main/deployments/helm/files/backup.sh) file needs to be modified also

**Review by eranreshef-starkware — CHANGES_REQUESTED** [review]

**Review by OmriEshhar1 — APPROVED** [review]
> Reviewed 4 of 4 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/1170)* status: all files reviewed, 1 unresolved discussion (waiting on @yoavGrs)

**Review by yoavGrs — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/1170)* status: 7 of 8 files reviewed, 1 unresolved discussion (waiting on @eranreshef-starkware)

___
*[`deployments/helm/templates/deployment.yaml` line 62 at r5](https://reviewable.io/reviews/starkware-libs/papyrus/1170#gh-1330202045:-NelMJwl5GtGDtTEIXL0:b-896fix) ([raw file](https://github.com/starkware-libs/papyrus/blob/83ee9ccd1d80271daaeb328052e4a12848aaf354/deployments/helm/templates/deployment.yaml#L62)):*
<details><summary><i>Previously, eranreshef-starkware (Eran Reshef) wrote…</i></summary><blockquote>

[backup.sh](https://github.com/starkware-libs/papyrus/blob/main/deployments/helm/files/backup.sh) file needs to be modified also
</blockquote></details>

Done.

**Review by eranreshef-starkware — COMMENTED** [review]
The changes LGTM but as you can see one of the CI stages fails.
The papyrus pod fails to start and what I see in the logs is:
```
Error: No such file or directory (os error 2)
```

> *[Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/1170#-:-Nem1L0pDGoaxKfgQuXW:bgbs57f)* status: 7 of 8 files reviewed, all discussions resolved (waiting on @yoavGrs)

**Review by OmriEshhar1 — APPROVED** [review]
> Reviewed 1 of 1 files at r6, 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/1170)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @yoavGrs)

**Review by OmriEshhar1 — CHANGES_REQUESTED** [review]
> Reviewed 4 of 4 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/1170)* status: all files reviewed, 1 unresolved discussion (waiting on @yoavGrs)

___
*[`deployments/helm/CI/values.yaml` line 5 at r7](https://reviewable.io/reviews/starkware-libs/papyrus/1170#-NemuHZX1Wdj4uaOG0_p:-NemuHZX1Wdj4uaOG0_q:b-bwmyku) ([raw file](https://github.com/starkware-libs/papyrus/blob/5ed62dfd3a6425e86a91b649cce828ab699e9219/deployments/helm/CI/values.yaml#L5)):*
> ```YAML
> rustLogLevel: "debug"
> 
> starknet:
> ```

was this reverted intentionally?

**Review by OmriEshhar1 — COMMENTED** [review]
> Reviewed 1 of 1 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/1170)* status: all files reviewed, 1 unresolved discussion (waiting on @yoavGrs)

**Review by yoavGrs — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/1170)* status: all files reviewed, 1 unresolved discussion (waiting on @OmriEshhar1)

___
*[`deployments/helm/CI/values.yaml` line 5 at r7](https://reviewable.io/reviews/starkware-libs/papyrus/1170#-NemuHZX1Wdj4uaOG0_p:-NfP_pkU3sImTK-cnO_s:bdwg7tr) ([raw file](https://github.com/starkware-libs/papyrus/blob/5ed62dfd3a6425e86a91b649cce828ab699e9219/deployments/helm/CI/values.yaml#L5)):*
<details><summary><i>Previously, OmriEshhar1 wrote…</i></summary><blockquote>

was this reverted intentionally?
</blockquote></details>

Yes,
The preset is an optional feature, and the CI should use defaults.

**Review by yoavGrs — COMMENTED** [review]
Fixed (in another PR).

> *[Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/1170#-:-NfPaBhh5KDnlokl0i1h:b-a1nf88)* status: all files reviewed, 1 unresolved discussion (waiting on @OmriEshhar1)

**Review by OmriEshhar1 — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/starkware-libs/papyrus/1170#-:-NfPaY4T4vTet46XI801:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @yoavGrs)

**Review by eranreshef-starkware — APPROVED** [review]

**Pull request merged**